### ISSUE TITLE
Snowstorm doesn't convert

### ISSUE THREAD

**Issue by Andreas-MJ** [graphnet-team/graphnet#412]
**Describe the bug**
I am trying to convert I3 files from the Snowstorm sample to sql. The I3-files contain 2 P-frames in a row, where only one has the specified pulsemap I need. I get the following error message:

```
(graphnet) [andreasmj@hep04 ~]$ python /groups/icecube/andreasmj/work/graphnet/examples/01_icetray/01_convert_i3_files.py sqlite icecube-86
graphnet: INFO     2023-02-01 15:27:15 - get_logger - Writing log to logs/graphnet_20230201-152715.log
graphnet: INFO     2023-02-01 15:27:34 - SQLiteDataConverter.info - Saving results to /groups/icecube/petersen/GraphNetDatabaseRepository/Snowstorm/merged
graphnet: INFO     2023-02-01 15:27:34 - SQLiteDataConverter.info - Processing 1623 I3 file(s) in main thread (not multiprocessing)
  0%|                                                                                                                   | 0/1623 [00:00<?, ?file(s)/s]graphnet: WARNING  2023-02-01 15:27:35 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:35 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:35 - warn_once - break_cyclic_recursion - Already evaluating object. Skipping recusion.
graphnet: WARNING  2023-02-01 15:27:35 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:35 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:35 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:36 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:36 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:36 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:37 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:37 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:37 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:37 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:37 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:37 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:38 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:38 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:38 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:42 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:42 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:42 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:42 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:42 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:42 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:46 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:46 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:46 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:46 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:46 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:46 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:47 - I3GenericExtractor.warning - Key LineFitParams not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:47 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:48 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:50 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:50 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:50 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:50 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:52 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:53 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:53 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
graphnet: WARNING  2023-02-01 15:27:53 - I3GenericExtractor.warning - Key SplitInIcePulses not in frame. Skipping
  0%|                                                                                                                   | 0/1623 [01:12<?, ?file(s)/s]
Traceback (most recent call last):
  File "/groups/icecube/andreasmj/work/graphnet/examples/01_icetray/01_convert_i3_files.py", line 113, in <module>
    main_icecube86(args.backend)
  File "/groups/icecube/andreasmj/work/graphnet/examples/01_icetray/01_convert_i3_files.py", line 48, in main_icecube86
    converter(inputs)
  File "/lustre/hpc/icecube/andreasmj/work/graphnet/src/graphnet/data/dataconverter.py", line 215, in __call__
    self.execute(filesets)
  File "/lustre/hpc/icecube/andreasmj/work/graphnet/src/graphnet/data/dataconverter.py", line 286, in execute
    pool = self._iterate_over_individual_files(filesets)
  File "/lustre/hpc/icecube/andreasmj/work/graphnet/src/graphnet/data/dataconverter.py", line 331, in _iterate_over_individual_files
    self._process_file, tqdm(args, unit="file(s)", colour="green")
  File "/lustre/hpc/icecube/andreasmj/work/graphnet/src/graphnet/data/dataconverter.py", line 83, in wrapper
    output_file = process_method(self, *args)
  File "/lustre/hpc/icecube/andreasmj/work/graphnet/src/graphnet/data/dataconverter.py", line 381, in _process_file
    self.save_data(data, output_file)
  File "/lustre/hpc/icecube/andreasmj/work/graphnet/src/graphnet/data/sqlite/sqlite_dataconverter.py", line 44, in save_data
    if len(dataframe[key]):
KeyError: 'SplitInIcePulses'
```

This is my script, based on the current example script in GraphNet

```
"""Example of converting I3-files to SQLite and Parquet."""

import logging
import os

from graphnet.constants import EXAMPLE_OUTPUT_DIR, TEST_DATA_DIR
from graphnet.data.extractors import (
    I3FeatureExtractorIceCubeUpgrade,
    i3featureextractor,
    I3RetroExtractor,
    I3TruthExtractor,
    I3GenericExtractor,
)
from graphnet.data.dataconverter import DataConverter
from graphnet.data.parquet import ParquetDataConverter
from graphnet.data.sqlite import SQLiteDataConverter
from graphnet.utilities.argparse import ArgumentParser
from graphnet.utilities.imports import has_icecube_package
from graphnet.utilities.logging import get_logger

logger = get_logger(level=logging.INFO)

CONVERTER_CLASS = {
    "sqlite": SQLiteDataConverter,
    "parquet": ParquetDataConverter,
}


def main_icecube86(backend: str) -> None:
    """Convert IceCube-86 I3 files to intermediate `backend` format."""
    # Check(s)
    assert backend in CONVERTER_CLASS

    inputs = [f"/groups/icecube/petersen/GraphNetDatabaseRepository/Snowstorm/"]
    outdir = f"/groups/icecube/petersen/GraphNetDatabaseRepository/Snowstorm/merged"

    converter: DataConverter = CONVERTER_CLASS[backend](
        [
            I3GenericExtractor(
                keys=[
                    "SplitInIcePulses",
                    "LineFitParams",
                ]
            ),
            I3TruthExtractor(),
        ],
        outdir,
    )
    converter(inputs)
    if backend == "sqlite":
        converter.merge_files(os.path.join(outdir, "merged"))


def main_icecube_upgrade(backend: str) -> None:
    """Convert IceCube-Upgrade I3 files to intermediate `backend` format."""
    # Check(s)
    assert backend in CONVERTER_CLASS

    inputs = [f"{TEST_DATA_DIR}/i3/upgrade_genie_step4_140028_000998"]
    outdir = f"{EXAMPLE_OUTPUT_DIR}/convert_i3_files/upgrade"
    workers = 1

    converter: DataConverter = CONVERTER_CLASS[backend](
        [
            I3TruthExtractor(),
            I3RetroExtractor(),
            I3FeatureExtractorIceCubeUpgrade("I3RecoPulseSeriesMap_mDOM"),
            I3FeatureExtractorIceCubeUpgrade("I3RecoPulseSeriesMap_DEgg"),
        ],
        outdir,
        workers=workers,
        # nb_files_to_batch=10,
        # sequential_batch_pattern="temp_{:03d}",
        # input_file_batch_pattern="[A-Z]{1}_[0-9]{5}*.i3.zst",
        icetray_verbose=1,
    )
    converter(inputs)
    if backend == "sqlite":
        converter.merge_files(os.path.join(outdir, "merged"))


if __name__ == "__main__":

    if not has_icecube_package():
        logger.error(
            "This example requires IceTray to be installed, which doesn't "
            "seem to be the case. Please install IceTray; run this example in "
            "the GraphNeT Docker container which comes with IceTray "
            "installed; or run an example script in one of the other folders:"
            "\n * examples/02_data/"
            "\n * examples/03_weights/"
            "\n * examples/04_training/"
            "\n * examples/05_pisa/"
            "\nExiting."
        )

    else:
        # Parse command-line arguments
        parser = ArgumentParser(
            description="""
Convert I3 files to an intermediate format.
"""
        )

        parser.add_argument("backend", choices=["sqlite", "parquet"])
        parser.add_argument(
            "detector", choices=["icecube-86", "icecube-upgrade"]
        )

        args = parser.parse_args()

        # Run example script
        if args.detector == "icecube-86":
            main_icecube86(args.backend)
        else:
            main_icecube_upgrade(args.backend)

```

**Comment by RasmusOrsoe** [comment]
@Andreas-MJ Try to use the i3FeatureExtractor instead of the generic one. Should work.

**Comment by asogaard** [comment]
@Andreas-MJ, can we close this issue? It seems like the error message in the [initial issue](https://github.com/graphnet-team/graphnet/issues/412#issue-1566253669) has been fixed by #432. I realise from #433 that this is not the end of your problems, but I think the best way to help you progress is to try and isolate any problems you might have, and then open dedicated issues for this.

**Comment by MortenHolmRep** [comment]
@asogaard @Andreas-MJ I am closing this issue. It is my understanding that @Andreas-MJ is not currently working on Snowstorm. The issue may remain for the specific dataset but needs more testing with the updated code. 
Feel free to reopen the issue.
